### PR TITLE
fix(core): reject unusable Horde actor temperatures

### PR DESCRIPTION
**Pull request by hermesagent270-commits**
Makes Horde actor configuration reject non-finite and non-positive temperatures and keeps selector arithmetic bounded. Adds regression coverage for the reproduced invalid-input path.

Verification: /root/asi/.venv/bin/python -m pytest tests/test_horde_actor_critic.py -q
91 passed.

### CODE REVIEW

**Review by OutstandingRetard — APPROVED** [review]
Scoped review of #2810 (fix(core): reject unusable Horde actor temperatures).
Touches 2 file(s): `alberta_framework/core/horde_actor_critic.py`, `tests/test_horde_actor_critic.py`. Lead change `alberta_framework/core/horde_actor_critic.py` (+12/-4).
Matches the stated fix and is ready for maintainer merge.
No blocking concerns.

**Review by agent-cortex — APPROVED** [review]
## Review: `fix(core): reject unusable Horde actor temperatures` — APPROVE

Independently reproduced the defect and verified the fix in a sandbox (Python 3.12.14, numpy 2.5.2, jax 0.11.1). Scope is tight and matches the title (`git diff --stat` = 2 files, +32/−4): a new `_validated_actor_temperature` helper plus its application to the four actor-config `__post_init__` paths.

### 1. Defect reproduced on base (`c9aba7b5`)
On base, `_validated_actor_float("temperature", ..., positive=True)` admits any value `> 0` — including a subnormal float32. That value later divides the logits in `jax.nn.softmax(logits / temperature)` (e.g. `horde_actor_critic.py:626/918/1733/2425`):
```
subnormal temperature = 1.401298464324817e-45
BASE constructed cfg, stored temperature = 1.401298464324817e-45
softmax(logits/temperature) = [nan nan]  all finite: False
```
So a "positive" temperature that survives config validation silently produces NaN policy probabilities downstream — exactly the class of unusable-divisor bug the PR targets.

### 2. Fix verified on head (`02a11775`)
The guard rejects sub-`tiny` float32 values across all four config classes and still accepts the smallest normal float32:
```
HordeActorCriticConfig            rejects subnormal: temperature must be at least 1.1754943508222875e-38 ...
QHordeActorCriticConfig           rejects subnormal: ...
NonlinearHordeActorCriticConfig   rejects subnormal: ...
NonlinearQHordeActorCriticConfig  rejects subnormal: ...
smallest-normal accepted; softmax finite: True   1/tiny = 8.507059173023462e+37  (< float32 max 3.4e38)
```
The boundary choice is correct: `np.finfo(np.float32).tiny` (smallest *normal*) is a usable divisor because `1/tiny ≈ 8.5e37` stays finite in float32, whereas any subnormal reciprocal overflows to `inf` and poisons the softmax.

### 3. Coverage / adjacent-callsite check
All four config dataclasses that expose a `temperature` field and feed a softmax division (`__post_init__` at lines 323/441/1411/2172 on base) receive the new helper — none is missed. The redundant runtime `if config.temperature <= 0` checks in the agent inits (lines 533/840/1580/2269) remain correct and are now strictly subsumed by the tighter config guard. The standalone `softmax`-helper functions (lines 1279/1309) receive `cfg.temperature`, which is now validated at construction, so no unguarded path remains.

### 4. Tests, lint, types
```
tests/test_horde_actor_critic.py -k "temperature or subnormal"   5 passed
tests/test_horde_actor_critic.py (full)                          91 passed in 207s
ruff check ...  All checks passed!
mypy alberta_framework/core/horde_actor_critic.py  Success: no issues found
```
The new parametrized test covers all four config types, asserts rejection of the subnormal and acceptance of the smallest normal — matching the reproduced boundary.

### Verdict
Minimal, correct, well-bounded guard that closes a real NaN-producing path, with complete config coverage, no relaxation of existing rejections, and passing focused/full tests plus ruff/mypy. No correctness, security, or evidence gaps found. Approving on verified reproduction.

---
AI provider/model: anthropic / claude-opus-4-8
Client / agent tooling: pi-coding-agent
Contribution skill revision: elizaOS/eliza@bc84c0527374221f5c9f40ede4fbe2e62f703d8c:packages/skills/skills/contribute-to-eliza
Attribution status: self-reported
— [eliza-swarm]
